test/IP3Token.EthlessTransfer.test.js

```diff
@@ -42,18 +42,54 @@ describe('IP3Token - Ethless Transfer functions', function () {
                 feeToPay,
                 nonce
             );
-            await IP3Token['transfer(address,address,uint256,uint256,uint256,bytes)'](
+            const input = await IP3Token.connect(user3).populateTransaction[
+                'transfer(address,address,uint256,uint256,uint256,bytes)'
+            ](owner.address, user2.address, amountToTransfer, feeToPay, nonce, signature);
+            await TestHelper.checkResult(input, IP3Token.address, user3, ethers, provider, 0);
+            expect(await IP3Token.balanceOf(owner.address)).to.equal(
+                ethers.BigNumber.from(originalBalance).sub(amountToTransfer).sub(feeToPay)
+            );
+            expect(await IP3Token.balanceOf(user2.address)).to.equal(ethers.BigNumber.from(amountToTransfer));
+            expect(await IP3Token.balanceOf(user3.address)).to.equal(ethers.BigNumber.from(feeToPay));
+        });
+    });
+
+    describe('IP3Token - Test expecting failure Ethless Transfer', async function () {
```

Comment:
it is the failed case when the nonce was used only, try to add tests when different fee and amount